test-apps/sediment/src/main/java/com/twosigma/waiter/sediment/Servlet.java

```diff
@@ -0,0 +1,222 @@
+package com.twosigma.waiter.sediment;
+
+import com.google.gson.Gson;
+import org.eclipse.jetty.http.HttpFields;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+
+import javax.servlet.ServletException;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class Servlet extends HttpServlet {
+
+    private final static Logger LOGGER = Logger.getLogger(Servlet.class.getName());
+
+    static String retrieveCorrelationId(HttpServletRequest request) {
+        final Enumeration<String> headerNames = request.getHeaderNames();
+        while (headerNames.hasMoreElements()) {
+            String headerName = headerNames.nextElement();
+            if ("x-cid".equalsIgnoreCase(headerName)) {
+                return request.getHeader(headerName);
+            }
+        }
+        return UUID.randomUUID().toString();
+    }
+
+    static long slurpRequest(final InputStream inputStream) {
+        long totalBytesRead = 0;
+        final byte[] buffer = new byte[32768];
+        try {
+            while (true) {
+                final int bytesRead = inputStream.read(buffer);
+                if (bytesRead == -1) {
+                    LOGGER.info("total bytes read = " + totalBytesRead);
+                    break;
+                }
+                totalBytesRead += bytesRead;
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return totalBytesRead;
+    }
+
+    @Override
+    protected void service(final HttpServletRequest request, final HttpServletResponse response)
+        throws IOException, ServletException {
+
+        final String correlationId = retrieveCorrelationId(request);
+        LOGGER.info(correlationId + " received " + request.getProtocol() + " " + request.getMethod() +
+            " request at " + request.getRequestURI());
+        response.setHeader("x-cid", correlationId);
+        response.setHeader("server", "sediment:" + HttpConfiguration.SERVER_VERSION);
+
+        final Request jettyRequest = (Request) request;
+        final Response jettyResponse = (Response) response;
+
+        switch (request.getRequestURI()) {
+            case "/":
```

Comment:
Let's remove "/" unless needed.  Do we need two endpoints that report status?

Reply:
Done.